### PR TITLE
[MINOR] improvement(all): Create a server-common module to reduce client-side dependencies

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes were proposed in this pull request?

Create a  server-common module.

### Why are the changes needed?

Reduce client-side denpendencies.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

No need

### CODE REVIEW

File: server/pom.xml [hunk 1]
```diff
@@ -36,6 +36,10 @@
       <groupId>org.apache.uniffle</groupId>
       <artifactId>shuffle-storage</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.uniffle</groupId>
+      <artifactId>rss-common</artifactId>
```
Comment (xianjingfeng) [inline server/pom.xml:41]: rss-server-common?

File: server-common/pom.xml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    <relativePath>../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>rss-server-common</artifactId>
+  <name>Apache Uniffle Server Common</name>
+
+  <properties>
+    <maven.compiler.source>8</maven.compiler.source>
+    <maven.compiler.target>8</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (xianjingfeng) [inline server-common/pom.xml:36]: Are these necessary?

Comment (maobaolong) [reply server-common/pom.xml]: No, it generated by idea, just remove it.

File: server/pom.xml (hunk 1 above)
Comment (maobaolong) [reply server/pom.xml]: done

**Comment by maobaolong** [comment]
@xianjingfeng Thanks for the review, addressed your comments, PTAL.

**Review by xianjingfeng — COMMENTED** [review]
LGTM. Wait for others.

**Comment by maobaolong** [comment]
@advancedxy Yes, I try to move web related code and dependencies into server-common, but I found that the client side start a ShuffleManagerGrpcService which dependent of web related third-party dependencies. 

So, I just start a first step, create a server-common module, then both I and @xianjingfeng can do some refactor base on this.

**Comment by maobaolong** [comment]
@advancedxy Done.

**Comment by rickyma** [comment]
Please change the description of this PR. The description will be treated as the git commit message.

**Comment by maobaolong** [comment]
@rickyma Thanks for your remind, updated the title and description. PTAL

**Comment by maobaolong** [comment]
@rickyma Thanks for your review!

**Pull request merged**